reporter/src/main/kotlin/utils/SpdxDocumentModelMapper.kt

```diff
@@ -211,6 +215,25 @@ private fun ScanResult.toSpdxPackageVerificationCode(): SpdxPackageVerificationC
         packageVerificationCodeValue = summary.packageVerificationCode.toLowerCase()
     )
 
+private fun SpdxDocument.addExtractedLicenseInfo(licenseTextProvider: LicenseTextProvider): SpdxDocument {
+    val nonSpdxLicenses = packages.flatMapTo(mutableSetOf()) {
+        SpdxExpression.parse(it.licenseConcluded).licenses() + SpdxExpression.parse(it.licenseDeclared).licenses()
```

Comment:
Hmm, is it really correct to use concluded *and* declared licenses here? At least in ORT's model, concluded trumps declared (and detected), so I would have expected to use concluded here, and only fall back to declared if there is no concluded license.

Reply:
I'm not following your argumentation why the fact that concluded trumps declared in ORT should imply that the license texts for the declared licenses do not need to be provided in an SpdxDocument (for packages with a concluded license).

Anyhow the SPDX specification says "One instance shall be created for every unique license or licensing information being referenced that does not match one of the licenses on the SPDX License List.", see https://github.com/spdx/spdx-spec/blob/development/v2.2.1/chapters/composition-of-an-SPDX-document.md#525-other-licensing-information-detected-.

I think the implementation `as-is` matches that specification and should not be changed. Do you agree @sschuberth ?

Reply:
Ah, to me the important bit to quote is "This section is used for any detected, declared or concluded licenses that are NOT on the SPDX License List". So aren't we then missing detected licenses here?

Reply:
Well, that's a different topic. But yes, detected licenses are missing. They are intentionally not yet implemented and will be added in a future iteration as it needs further clarifications.

Reply:
@sschuberth  does it resolve this discussion?

Reply:
As it is completely not obvious from neither the commit message nor the code that you're aware that the implementation is not complete, please change the commit message to

```
The extracted license info list is supposed to contain "any detected,
declared or concluded licenses that are NOT on the SPDX License List".
Only implement this for concluded and declared licenses for now as
also getting detected licenses requires more work.
```

and add

```
// TODO: Also add detected non-SPDX licenses here.
```

before line 220 to the code.

Reply:
Done.